ha_healthchecker/README.md

```diff
@@ -0,0 +1,146 @@
+
+        b. If the service is marked as `down`, `ha_healthchecker` will send a github issue to `openlab` repo and try to start switching Master and Slave.
+
+    2. If the service is an unnecessary service.
+
+        a. If the service is marked as `restarting`, it'll be restarted as well.
+
+        b. If the service is marked as `down` but the the time doesn't hit the switch deadline, `ha_healthchecker` will only send out a github issue.
+
+        c. If the service is marked as `down` and the the time hits the switch deadline, `ha_healthchecker` try to start switching Master and Slave.
+
+* **Switch**
+
+    Once **Fix** tell that OpenLab HA deployment need switch Master and Slave, and the configuration `allow_switch` is `True`. `ha_healthchecker` will send a github issue to `openlab` repo and do the switch work.
+
+## Configuration
+
+There are some configurations that used by `ha_healthchecker`. print `openlab ha config list` to get the value.
+
+| Name | Default Value | Description |
+| ---- | ------------- | ----------- |
+| allow_switch | False | Whether allow switch the HA deployment or not. |
+| dns_log_domain | test-logs.openlabtesting.org | The log service domain name. Usually it's `logs.openlabtesting.org`. |
+| dns_master_public_ip | None | The IP of the master node that runs zuul-web. |
+| dns_provider_account | None | The dns server account  for login. OpenLab use simpleDNS by default. |
+| dns_provider_api_url | https://api.dnsimple.com/v2/ | The dns server API url for login. |
+| dns_provider_token | None | The dns server token for login. |
+| dns_slave_public_ip | None | The IP of the slave node that runs zuul-web.  |
+| dns_status_domain | test-status.openlabtesting.org' | The zuul status web domain name. Usually it's `status.openlabtesting.org`. |
+| github_app_name | None | The github app that used by openlab CI, Usually it's `theopenlab-ci`. |
+| github_repo | None | The github repo that used by openlab CI, Usually it's `theopenlab/openlab`. |
+| github_user_name | None | The user name used to login github. |
+| github_user_password | None | The password used to login github. |
+| github_user_token | None | The token used to login github. |
+| heartbeat_timeout_second | 600 | How long the node is treated as down once the heartbeat won't be refreshed. |
+| logging_level | DEBUG | The log level for `ha_healthchecker` itself. |
+| service_restart_max_times | 3 | How many times that the service will be restarted once it's broken. |
+| unnecessary_service_switch_timeout_hour | 48 | How long the switch will be happened once an unnecessary service is down. |
+
+## How to use
+
+Once OpenLab HA deployment is running, print `openlab ha node list` and `openlab ha service list`, you can find the HA cluster's status.
+
+Usually Both nodes and services should be in `up` status. If any is not in `up` status, you should take a look at the github issue to see what's happened.
```

Comment:
github issue link better to be added, and the issue tittle prefix

Reply:
Sure